### PR TITLE
Upgrade for Coq 8.11

### PR DESCRIPTION
**Pull request by Tuplanolla**
I wanted to upgrade this plugin to work with Coq version 8.11.
However, I have no idea what I am doing, so,
while this attempt compiles and imports properly,
it fails with syntax errors on `ne_collection_list`.
I also applied some violence to `tclIDTAC_MESSAGE`,
which loses an error message along the way; sorry about that.

It would nice if you could finish what I started here,
because this plugin seems quite useful.

### CODE REVIEW

File: src/g_ltac_iter.mlg [hunk 1]
```diff
@@ -76,11 +80,11 @@ struct
             Hints.Hint_db.iter (fun _ _ hl -> syms := hl :: !syms) db ;
             !syms
           with Not_found ->
-            let _ = Tacticals.tclIDTAC_MESSAGE Pp.(str "Hint database " ++ str db_name ++ str " not found!") in
+            (* let _ = Tacticals.tclIDTAC_MESSAGE Pp.(str "Hint database " ++ str db_name ++ str " not found!") in *)
```
Comment (gmalecha) [inline src/g_ltac_iter.mlg]: Perhaps the right thing to do here is simply use the feedback mechanism.

```ocaml
Feedback.msg_warning Pp.(str "Hint database " ++ str db_name ++ str " not found!") ;
```

I guess if you want to be good, you should record the location information and report it as well.

File: src/g_ltac_iter.mlg [hunk 2]
```diff
@@ -154,28 +158,29 @@ let pp_collection _ _ _ = WITH_DB.pr_collection
...
-PRINTED BY pp_collection
-  | [ "db:" preident(x) ] -> [ WITH_DB.HintDb x ]
-  | [ "mod:" constr(m) ]  -> [ WITH_DB.Module m ]
-  | [ "*|-" ] -> [ WITH_DB.Premise ]
-  | [ "rev" collection(e) ] -> [ WITH_DB.Reverse e ]
-  | [ "ctors:" constr(t) ] -> [ WITH_DB.Ctors t ]
-  | [ "(" collection(e) ")" ] -> [ e ]
-END;;
+PRINTED BY { pp_collection }
+  | [ "db:" preident(x) ] -> { WITH_DB.HintDb x }
```
Comment (gmalecha) [inline src/g_ltac_iter.mlg:166]: Is there any way to get the lexing location here? That would allow us to print it above.

Comment (Tuplanolla) [reply src/g_ltac_iter.mlg:166]: Unfortunately, I do not have the OCaml skills needed to do this properly.

**Comment by gmalecha** [comment]
Thanks! @Tuplanolla

**Comment by Tuplanolla** [comment]
Note that this still fails with syntax errors on `ne_collection_list`.

**Pull request merged**